sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -228,6 +229,12 @@ final class Decimal extends Ordered[Decimal] with Serializable {
     if (decimalVal.eq(null)) {
       longVal / POW_10(_scale)
     } else {
+      if (SQLConf.get.arithmeticOperationOverflowCheck) {
```

Comment:
shall we use `DECIMAL_OPERATIONS_NULL_ON_OVERFLOW` config instead?

Reply:
BTW we can do it in another PR, let's focus on non-decimal arithmetic operation in this PR.

Reply:
And we need to handle more cases as well, e.g. long to int, int to byte, etc. AFAIK @gengliangwang is working on cast recently.

Reply:
IIRC, I had to do this in order to have some test cases work. IIRC it is related to the sum buffer. I'd consider quite challenging decide the right config among these 2 honestly.... It may be quite counter-intuitive for a user that a sum of long can be affected by `DECIMAL_OPERATIONS_NULL_ON_OVERFLOW `...

Reply:
> I had to do this in order to have some test cases work.

Not anymore if we revert https://github.com/apache/spark/pull/21599#discussion_r308722618 ?

Reply:
yes, exactly. We can discuss on this when we do the PR for addressing the sum buffer datatype.